hpx/lcos/future.hpp

```diff
@@ -140,9 +138,8 @@ namespace hpx { namespace lcos { namespace detail
         boost::is_void<typename traits::future_traits<Future>::type>
     >::type serialize_future_save(Archive& ar, Future const& f) //-V659
     {
-        if(f.valid())
+        if (f.valid())
         {
-            HPX_ASSERT(f.is_ready());
```

Comment:
Because this function might get called when the future isn't ready yet, as per the removal of `invoke_when_ready`